src/frontend/src/flows/authenticate/postMessageInterface.ts

```diff
@@ -36,6 +37,60 @@ export interface AuthRequest {
   derivationOrigin?: string;
 }
 
+/** Try to read unknown data as authentication request */
+const asAuthRequest = (msg: unknown): AuthRequest | undefined => {
+  if (typeof msg !== "object") {
+    return undefined;
+  }
+
+  if (msg === null) {
+    return undefined;
+  }
+
+  // Some extra conversions to take typescript by the hand
+  // eslint-disable-next-line
+  const tmp: {} = msg;
+  const obj: Record<string, unknown> = tmp;
```

Comment:
Why not just `const obj = msg as Record<string, unknown>;`

Reply:
Because in this case it's not assigning anymore, but it's casting, meaning you could even do that:

```ts
const obj = msg as AuthRequest;
```